src/Stratis.Bitcoin.Features.BlockStore/BlockStore.cs

```diff
@@ -0,0 +1,363 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Base;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.BlockStore
```

Comment:
`BlockStore` class in namespace `BlockStore` doesn't feel right

Reply:
It's kinda ok. We have several classes like that. For instance- namespace .Wallet has Wallet class

Reply:
As a hierarchical organisation of things, it doesn't seem right but as you say, the Feature is called BlockStore so you're kinda stuck

Reply:
`BlockStoreLoop` `BlockStoreScheduler` `BlockStoreHandler` `` etc... we have options.

Reply:
I don't mind changing the name. How about `BlockStoreQueue` or `BlockStoreManager`? 

>`BlockStoreLoop`

there is no loop anymore

> `BlockStoreHandler` 

There is a handler pattern so probably this name might be missleading for some people

> `BlockStoreScheduler`

That one or BlockStoreBatchScheduler sound good to me.

Reply:
Yes I like `BlockStoreQueue` lets do that it implies this is a store and a queue which is just that.

Reply:
Fixed